patches/api/0343-Add-Shearable-API.patch

```diff
@@ -0,0 +1,72 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Sun, 17 Oct 2021 15:39:41 -0400
+Subject: [PATCH] Add Shearable API
+
+
+diff --git a/src/main/java/io/papermc/paper/entity/Shearable.java b/src/main/java/io/papermc/paper/entity/Shearable.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..7411c389aadb8c587d5653219b99d6b4899db453
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/Shearable.java
+@@ -0,0 +1,21 @@
++package io.papermc.paper.entity;
++
++import net.kyori.adventure.sound.Sound;
++import org.jetbrains.annotations.NotNull;
++
++public interface Shearable {
++
++    /**
++     * Causes the entity to play the effect of being shared by a player.
```

Comment:
If `readyFoShearing` is true, then calling this method will actually shear the Shearable in addition to playing the sound. This should be noted in the Javadoc.